vyper/venom/analysis/mem_ssa.py

```diff
@@ -0,0 +1,388 @@
+            return self._get_in_def(idom) if idom else self.live_on_entry
+
+        return self.live_on_entry
+
+    def _get_reaching_def_for_def(self, bb: IRBasicBlock, def_inst: MemoryDef) -> MemoryAccess:
+        """Get the reaching definition for a memory definition"""
+        def_idx = bb.instructions.index(def_inst.store_inst)
+        def_loc = def_inst.loc
+
+        # First check for any previous memory def in the same basic block
+        for inst in reversed(bb.instructions[:def_idx]):
+            if inst in self.inst_to_def:
+                prev_def = self.inst_to_def[inst]
+                # Get most recent memory def, regardless of aliasing
+                return prev_def
+
+        if bb in self.memory_phis:
+            phi = self.memory_phis[bb]
+            for op, _ in phi.operands:
+                if isinstance(op, MemoryDef) and self.alias.may_alias(def_loc, op.loc):
+                    return phi
+
+        if bb.cfg_in:
+            idom = self.dom.immediate_dominators.get(bb)
+            if idom:
+                in_def = self._get_in_def(idom)
+                # Only use the in_def if it might alias with our definition
+                if isinstance(in_def, MemoryDef) and self.alias.may_alias(def_loc, in_def.loc):
+                    return in_def
+
+        return self.live_on_entry
+
+    def _remove_redundant_phis(self):
+        """Remove unnecessary phi nodes"""
+        for phi in list(self.memory_phis.values()):
+            if all(op[0] == phi for op in phi.operands):
+                del self.memory_phis[phi.block]
+
+    def get_clobbered_memory_access(self, access: Optional[MemoryAccess]) -> Optional[MemoryAccess]:
+        if access is None or access.is_live_on_entry:
+            return None
+
+        query_loc = access.loc
+        if isinstance(access, MemoryPhi):
+            # For a phi, check all incoming paths
+            for acc, _ in access.operands:
+                clobbering = self._walk_for_clobbered_access(acc, query_loc)
+                if clobbering and not clobbering.is_live_on_entry:
+                    # Phi itself if any path has a clobber
+                    return access
+            result = self.live_on_entry
+        else:
+            result = (
+                self._walk_for_clobbered_access(access.reaching_def, query_loc)
+                or self.live_on_entry
+            )
+
+        return result
+
+    def _walk_for_clobbered_access(
+        self, current: Optional[MemoryAccess], query_loc: MemoryLocation
+    ) -> Optional[MemoryAccess]:
+        while current and not current.is_live_on_entry:
+            if isinstance(current, MemoryDef) and query_loc.completely_overlaps(current.loc):
+                return current
+            elif isinstance(current, MemoryPhi):
+                for access, _ in current.operands:
+                    clobbering = self._walk_for_clobbered_access(access, query_loc)
+                    if clobbering:
```

Comment:
please, no truthy as it obscures the type --
```suggestion
                    if clobbering is not None:
```